packages/app/src/cli/utilities/host-theme-manager-next.ts

```diff
@@ -0,0 +1,37 @@
+import {waitForThemeToBeProcessed} from './host-theme-watcher.js'
+import {getHostTheme, removeHostTheme, setHostTheme} from '@shopify/cli-kit/node/themes/conf'
+import {ThemeManager} from '@shopify/cli-kit/node/themes/theme-manager'
+import {AdminSession} from '@shopify/cli-kit/node/session'
+import {createHostTheme} from '@shopify/cli-kit/node/themes/api'
+import {BugError} from '@shopify/cli-kit/node/error'
+import {DEVELOPMENT_THEME_ROLE} from '@shopify/cli-kit/node/themes/utils'
+import {generateThemeName} from '@shopify/cli-kit/node/themes/generate-theme-name'
+
+export class HostThemeManager extends ThemeManager {
```

Comment:
Personally, I believe the `theme-app-extension-next.ts` is a fair approach because it's our new entry point.

Still, I wonder if having many `*-next.ts` files will make our lives difficult when we need to clean up the legacy code. Thus, I believe an approach like this may be easier to clean up in the future:

```ts
async function findOrCreateHostTheme(adminSession: AdminSession): Promise<Theme> {
  const themeManager = new HostThemeManager(adminSession, { devPreview: true })
  const theme = await themeManager.findOrCreate()
  return theme
}
```

What do you think? Feel free to keep it as is if you feel strong about the current approach (I understand the `ThemeManager` holds a bit of complexity, so both alternatives have their trade-offs)